jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -996,28 +992,27 @@ def _update_stretch_histogram(self, msg={}):
                                 (y_data >= viewer.state.y_min) &
                                 (y_data <= viewer.state.y_max))
 
-                sub_data = comp.data[inds].ravel()
+                sub_data = comp.data[inds]
 
         else:
-            if self.config == "imviz":
-                # Downsample input data to about 400px (as per compass.vue) for performance.
-                xstep = max(1, round(data.shape[1] / 400))
-                ystep = max(1, round(data.shape[0] / 400))
-                arr = comp[::ystep, ::xstep]
-            else:
-                # include all data, regardless of zoom limits
-                arr = comp.data
-            sub_data = arr.ravel()
-
-        # filter out nans (or else bqplot will fail)
-        if np.any(np.isnan(sub_data)):
-            sub_data = sub_data[~np.isnan(sub_data)]
+            # include all data, regardless of zoom limits
+            sub_data = comp.data
 
+        self.stretch_histogram.viewer.state.random_subset = RANDOM_SUBSET_SIZE
         self.stretch_histogram._update_data('histogram', x=sub_data)
 
         if len(sub_data) > 0:
-            interval = PercentileInterval(95)
-            hist_lims = interval.get_limits(sub_data)
+
+            # Use glue to compute the statistics since this allows us to use
+            # a random subset of the data to compute the histogram
+            glue_data = self.stretch_histogram.app.data_collection['histogram']
+            hist_lims = (
+                glue_data.compute_statistic('percentile', glue_data.id['x'],
+                                            percentile=2.5, random_subset=RANDOM_SUBSET_SIZE),

```

Comment:
I don't have strong opinion here. I always think of this histogram as quick look and should not be taken too seriously but @camipacifici or @kecnry might have different opinions, so let's see if they comment.

Reply:
I do not know what this does exactly, so not sure how to comment, sorry.

Reply:
I think this is something we'll want to deal with eventually since it isn't always ideal, but this was hardcoded before so I think its reasonable to keep the same assumption for now